apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -326,11 +298,13 @@ class LivechatClass {
 
 		this.logger.debug(`DB updated for room ${room._id}`);
 
+		const transcriptRequested =
+			!!transcriptRequest || (!settings.get('Livechat_enable_transcript') && settings.get('Livechat_transcript_send_always'));
```

Comment:
Wouldn't it be better to check the "always" one first?

Reply:
This is not checking, i'm not sure what we use this value for tbh but we pass `transcriptRequested: true|false` to the closing message.

In here, if we already have a transcript request (via the `transcriptRequest` object) then we assume it's true.

BUT, if we don't have a transcriptRequest object but the setting is enabled, we assume this as `true` as well because a transcript will be sent.

TLDR: that thing is just a flag and we can check any of those first.